lib/schemas/src/hooks/circleci.ts

```diff
@@ -3,65 +3,79 @@ import { z } from 'zod'
 export const CircleCiCustom = z.record(z.unknown())
 export type CircleCiCustom = z.infer<typeof CircleCiCustom>
 
-export const CircleCiExecutor = z.object({
-  name: z.string(),
-  image: z.string()
-})
+export const CircleCiExecutor = z
+  .object({
+    name: z.string(),
+    image: z.string()
+  })
+  .partial()
+  .required({ name: true })
```

Comment:
**question**: what's the difference between this and doing `image: z.string().optional()`?

Reply:
I want to preserve the semantics of the original schema – `image` is still required for the final merged schema, just not required in a single plugin's hook options. We don't do any validation on the final schema at the moment so I think it makes sense to call `.partial` directly on the schema but in the future they could be split out.

Reply:
oh gotcha, i think i understand